### PR TITLE
Allow 00 based international syntax

### PR DESCRIPTION
**Pull request by mixman**

### CODE REVIEW

File: sms-proxy/package.yaml [hunk 1]
```diff
@@ -17,17 +17,18 @@ library:
...
     - futurice-prelude
     - futurice-servant
-    - servant-server
     - http-client
     - http-client-tls
     - http-types
-    - aeson
-    - aeson-compat
-    - aeson-extra
+    - regex-compat
```
Comment (phadej) [inline sms-proxy/package.yaml]: ei

File: sms-proxy/src/Futurice/App/SmsProxy/Logic.hs [hunk 2]
```diff
@@ -12,23 +12,27 @@ import Futurice.App.SmsProxy.Ctx
...
 
+import Text.Regex (subRegex, mkRegex)
+import qualified Data.Text as T
+
 sendSms :: (MonadIO m, MonadLog m) => Ctx -> Req -> m Res
 sendSms ctx req = do
-    -- Log "metadata"
     logInfo_ $ "Sending message to " <> req ^. reqTo
 
+    let number = T.pack $ subRegex (mkRegex "^00") (T.unpack $ req ^. reqTo) "+"
```
Comment (phadej) [inline sms-proxy/src/Futurice/App/SmsProxy/Logic.hs]: use either http://hackage.haskell.org/package/regex-applicative-text-0.1.0.1/docs/Text-Regex-Applicative-Text.html
or `T.isPrefixOf`, `T.drop 2` and `<>`.

Comment (phadej) [inline sms-proxy/src/Futurice/App/SmsProxy/Logic.hs]: FWIW, grep the repo, there are uses of `T.R.A.Text`.

File: sms-proxy/package.yaml (hunk 1 above)
Comment (mixman) [inline sms-proxy/package.yaml]: Routine sort needs sort -i; didn't even look back at the results.

File: sms-proxy/src/Futurice/App/SmsProxy/Logic.hs (hunk 2 above)
Comment (mixman) [inline sms-proxy/src/Futurice/App/SmsProxy/Logic.hs]: What's the benefit of the different RE package? Better suited for the types used?

Comment (phadej) [inline sms-proxy/src/Futurice/App/SmsProxy/Logic.hs]: no syntax errors in re syntax. And no reason to use many libraries for the same thing.

**Comment by phadej** [comment]
I'll deploy this shortly (will build `planmill-proxy` anyway too soon).

**Pull request merged**